File: solutions/3028. Ant on the Boundary/3028.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
public:
    int returnToBoundaryCount(const vector<int>& nums) {
        int count = 0;
        int pos = 0;
        
        for (int num : nums) {
            pos += num;
            if (pos == 0) {
                ++count;
            }
        }
        
        return count;
    }
};

File: solutions/3028. Ant on the Boundary/3028.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
    def returnToBoundaryCount(self, nums: List[int]) -> int:
        count = 0
        pos = 0

        for num in nums:
            pos += num
            if pos == 0:
                count += 1

        return count